### PR TITLE
Add decorate class

### PR DESCRIPTION
**Pull request by codecaiine**
- Create a class Corrector with a method correct_name that does the following:
- Make sure the first letter of the given name is a capital letter.
- Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.
- Set an instance of Corrector in Person on initialization.
- In Person add a method validate_name that uses the Corrector instance to validate and save the corrected name.

**This is my last review request. I implemented all request changes from the previous request reviews please reach out to me on slack if anything's wrong**

### CODE REVIEW

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Corrector
+  def correct_name(name)
+    given_name = name.strip.capitalize
+    if given_name.length > 10
+      given_name[0, 10]
+    else
+      given_name
+    end
+  end
+end
```
Comment (mikethreels) [inline corrector.rb:10]: - [x] good job setting up the `corrector` class 👍 however in the `Person` there should be an instance of Corrector in Person on initialization set.
- [x] In the `Person` class there should be a method validate_name that uses the Corrector instance to validate and save the corrected name.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Yannick, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by codecaiine** [comment]
Hi @mikethreels,
Thank you for reviewing my works. I did not notice that I haven't pushed all the code. 

Cheers!

File: corrector.rb (hunk 1 above)
Comment (henatan99) [reply corrector.rb:10]: - [x] Very nice work addressing the issue raised here. Unfortunately, you forgot to import the `Corrector` class in the `Person` class. That basically means the `Person` class cannot access the `Corrector` class and trying to initialize an object instance of a `Corrector` class will raise an error. Kindly fix this issue. :pray: 

Screenshot from 2021-11-16 12-41-39

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @codecaiine , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by codecaiine** [comment]
Hi @henatan99, 
Thank you for reviewing my work. But can you tell me more about the request changes you talked about in your comment?

Cheers!

**Comment by codecaiine** [comment]
Hi @henatan99 ,

Thank you for reviewing my work. I implemented all required changes from previous and your comments. As you can confirm on these screenshots below. 
demmmmmmmm
demoopooooo

grgrgrgrgrgrgrgrgrgrgrgrgrgr

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @codecaiine 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by codecaiine** [comment]
Hi @helman101, 

Thank you for reviewing my works and approving it. 

Cheers!

**Pull request merged**